### PR TITLE
Add webhook resources in git import pipeline flow

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-6394

**Analysis / Root cause**: 

As a user, while importing an application (git import flow), I want to have a default webhook setup generated and provide the instructions to configure the webhook url in the git provider.

**Solution Description**: 

1. Added the webhook resources, if the user checks the Add pipeline option
2. ~Show toast message with webhook Url~ (removed after discussing with UX)
3. Show the Webhook resources and Url in the Sidepanel of the selected resources.

**Screen shots / Gifs for design review**: 

Sidebar Entry for Triggers:
image

GIF:
https://user-images.githubusercontent.com/9964343/146979399-4bfd6fcd-a288-4a53-8a30-af81aff7b5d0.mov

**Unit test coverage report**: 

```
 createPipelineResource tests
      ✓ should create pipeline resource with same name as application name (3ms)
      ✓ should suppress the error if the trigger creation fails with the error (1ms)
      
  Pipeline sidebar overview
    ✓ should not show trigger template and url when there are no templateNames (33ms)
    ✓ should show trigger template and url (30ms)   
```
**Test setup:**

1. Add deployment through git import flow
2. Check `Add Pipeline` option.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc:  @openshift/team-devconsole-ux  @beaumorley @serenamarie125 @siamaksade

### CODE REVIEW

**Comment by beaumorley** [comment]
@karthikjeeyar 
I am curious as to why we choose to use toasts for this?  I know we are using them for the export flow. But this seems more similar to Health Checks to me (something that is part of your import parameters) and we use a notification on the side panel for those (when you have not set them). Wondering if there was a UX discussion with @serenamarie125  on this and if and how to notify the user? I know she is out. I am happy to review any notes if there are any. I looked in the OCD/PM/UX sync agenda and did not see anything. I love the plan to use defaults and set this up. Just want to make sure the level of notification we give the user is right. Thanks!

File: frontend/packages/dev-console/src/components/import/ImportForm.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { RouteKind } from '@console/internal/module/k8s';
 import { getActiveApplication } from '@console/internal/reducers/ui';
 import { RootState } from '@console/internal/redux';
 import { KnativeServingModel } from '@console/knative-plugin/src';
+import { addWebhookToast } from '@console/pipelines-plugin/src/components/import/pipeline/webhook-utils';
```
Comment (abhinandan13jan) [inline frontend/packages/dev-console/src/components/import/ImportForm.tsx:15]: should we move this to topology instead?

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/webhook-utils.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      <p>
+        <RouteLinkAndCopy route={route} />
+      </p>
+    </>
+  ),
+  timeout: true,
+  dismissible: true,
+});
+
+export const addWebhookToast = async (resources: K8sResourceKind[], toastContext) => {
```
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/webhook-utils.tsx:24]: should we add type toastContextType

Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/webhook-utils.tsx]: This file is removed, as we dont need toast at the moment (discussed with UX on this).

**Comment by karthikjeeyar** [comment]
> @karthikjeeyar I am curious as to why we choose to use toasts for this? I know we are using them for the Export flow. But this seems more similar to Health Checks to me (something that is part of your import parameters) and we use a notification on the side panel for those (when you have not set them). Wondering if there was a UX discussion with @serenamarie125 on this on how to notify the user? I know she is out. I am happy to review any notes if there are any. I looked in the OCD/PM/UX sync agenda and did not see anything. I love the plan to use defaults and set this up. Just want to make sure the level of notification we give the user is right. Thanks!

@beaumorley As per our discussion in slack, I have removed the toast message for now. Thanks for the review.

**Comment by beaumorley** [comment]
LGKM. I think there is still a legacy .mov file ^ that shows the toast. Can we delete that? I'll take the an action item to p/u discussion on notifications with PM in January.

**Comment by abhinandan13jan** [comment]
/lgtm 
ran locally. works as expected

**Comment by karthikjeeyar** [comment]
Propagating docs and px acks from the [Epic](https://issues.redhat.com/browse/ODC-6375) 
/label docs-approved
/label px-approved

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 3]
```diff
@@ -476,6 +478,14 @@ export const managePipelineResources = async (
       docker.dockerfilePath,
       image.tag,
     );
+    pipelineResources.push(managedPipeline);
+    try {
+      const res = await createTrigger(managedPipeline, git.detectedType);
+      if (res && res.length > 0) pipelineResources.push(...res);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log('Error occured while creating triggers', error);
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:487]: nit: 
```suggestion
      console.warn('Error occured while creating triggers', error);
```

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/TriggerResourceLinks.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.odc-trigger-template-link {
```
Comment (invincibleJai) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/TriggerResourceLinks.scss:1]:
```suggestion
.opp-trigger-template-link {
```

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 5]
```diff
@@ -476,6 +478,14 @@ export const managePipelineResources = async (
       docker.dockerfilePath,
       image.tag,
     );
+    pipelineResources.push(managedPipeline);
+    try {
+      const res = await createTrigger(managedPipeline, git.detectedType);
+      if (res && res.length > 0) pipelineResources.push(...res);
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:484]: is this needed?

Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: Not needed, removed.

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 6]
```diff
@@ -476,6 +478,14 @@ export const managePipelineResources = async (
       docker.dockerfilePath,
       image.tag,
     );
+    pipelineResources.push(managedPipeline);
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:481]: `managedPipeline` can be undefined right if above if and else if is not truthy so here shall we have check?

```suggestion
    managedPipeline && pipelineResources.push(managedPipeline);
```

Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts:481]: managePipelineResources is being called when a pipeline is enabled, so managedPipeline here will be always set to either create flow or update flow.

**Pull request merged**